common/build_data_linkstamp.h

```diff
@@ -19,12 +17,13 @@ namespace Carbon::BuildData::Internal {
 // use).
 //
 // Also, when build_data_linkstamp.cpp is compiled, this doesn't receive deps,
-// so we can't use things like `llvm::StringRef` here. As a result, we use
+// so we can't use things like `llvm::StringRef` here. It should ideally be
+// purely hermetic -- not even using STL for `string_view`. As a result, we use
 // `build_data.h` as an intermediary to do a `StringRef` wrap.
-extern const std::string_view platform;
+extern const char platform[];
```

Comment:
What do you think about manually exposing a length global as well, and using that to construct the `StringRef` inline variables without running `strlen`? Should be able to write this in the `.cpp` file without any headers by using `__builtin_strlen` to initialize, or `sizeof(...) - 1` ... not sure which is better/worse.

Is that too much magic?

Reply:
I'm wary of the extra complexity that adds. I'd prefer for it to be offset by a performance benefit, but I wouldn't expect one here. Do you think it's going to be significant?

Reply:
It's not so much a performance benefit as trying to make the `StringRef` not have a dynamic initializer that has to be emitted into every file that uses it (and then merged by the linker). I don't think it'll be too much complexity, but happy to send it in a follow-up PR if you'd rather.

Reply:
If the amount of code emitted is your concern, how about this -- I've split out build_data.cpp so that users don't see initialization. Note, this mirrors how the main other implementation I'm aware of is handling it.

Reply:
Well, kind of. To mirror in full would be to switch to functions returning the values. It's not obvious to me what trade-off they're choosing for that.

Reply:
Eh, it avoids duplicating the code, but will also make uses of these marginally less efficient. Maybe that doesn't matter.

What I really wish is that we could have dependencies in linkstamp so these were `constexpr` `StringLiteral`s. Oh well.

Anyways, it seems better to just get something landed that sorts out build issues rather than worrying about this in this iteration.

Reply:
To be sure, for efficiency concerns, this library is currently used in file_test_base_test.cpp; I'm not aware of other planned uses.